### PR TITLE
Minor static analysis fixes

### PR DESCRIPTION
**Pull request by kaduk**
Coverity spotted a few potential issues in our codebase based off of 1.0.2; a couple of them still apply to master, so I'm sending them in.

### CODE REVIEW

File: crypto/bio/bio_cb.c [hunk 1]
```diff
@@ -77,6 +77,7 @@ long BIO_debug_callback(BIO *bio, int cmd, const char *argp,
 
     len = BIO_snprintf(buf,sizeof buf,"BIO[%p]: ",(void *)bio);
 
+    len = (len < 0) ? 0 : len;
```
Comment (richsalz) [inline crypto/bio/bio_cb.c]: make this "if len < 0 len = 0" with a comment about ignoring errors.

Comment (kaduk) [inline crypto/bio/bio_cb.c]: updated, and rebased on current master.

**Comment by richsalz** [comment]
+1 will merge once someone else reviews.

**Comment by levitte** [comment]
+1

**Comment by richsalz** [comment]
merged, closing; thanks.

**Pull request closed without merging**